mlflow/transformers.py

```diff
@@ -0,0 +1,919 @@
+
+    flavor_config = _get_flavor_configuration_from_uri(model_uri, FLAVOR_NAME, _logger)
+
+    if return_type == "pipeline" and _PROCESSOR_TYPE_KEY in flavor_config:
+        raise MlflowException(
+            "This model has been saved with a processor. Processor objects are "
+            "not compatible with Pipelines. Please load this model by specifying "
+            "the 'return_type'='components'.",
+            error_code=BAD_REQUEST,
+        )
+
+    _add_code_from_conf_to_system_path(local_model_path, flavor_config)
+
+    components = _load_model(local_model_path, flavor_config)
+    if return_type == "pipeline":
+        components.update(**kwargs)
+        return transformers.pipeline(**components)
+    elif return_type == "components":
+        return components
+
+
+def _load_model(path: str, flavor_config):
+    """
+    Loads components from a locally serialized ``Pipeline`` object.
+    """
+    import transformers
+
+    local_path = pathlib.Path(path)
+    pipeline_path = local_path.joinpath(
+        flavor_config.get(_PIPELINE_BINARY_KEY, _PIPELINE_BINARY_FILE_NAME)
+    )
+
+    model_instance = getattr(transformers, flavor_config[_PIPELINE_MODEL_TYPE_KEY])
+    conf = {
+        "task": flavor_config[_TASK_KEY],
+        "model": model_instance.from_pretrained(pipeline_path),
+    }
+
+    if _PROCESSOR_TYPE_KEY in flavor_config:
+        conf[_PROCESSOR_KEY] = _load_component(
+            local_path, _PROCESSOR_KEY, flavor_config[_PROCESSOR_TYPE_KEY]
+        )
+
+    for component_key in flavor_config[_COMPONENTS_BINARY_KEY]:
+        component_type_key = f"{component_key}_type"
+        component_type = flavor_config[component_type_key]
+        conf[component_key] = _load_component(local_path, component_key, component_type)
+    return conf
+
+
+def _fetch_model_card(model_or_pipeline):
+    """
+    Attempts to retrieve the model card for the specified model architecture iff the
+    `huggingface_hub` library is installed. If a card cannot be found in the registry or
+    the library is not installed, returns None.
+    """
+    try:
+        hub = importlib.import_module("huggingface_hub")
+        model = model_or_pipeline.model
+
+        if hasattr(hub, "ModelCard"):
+            return hub.ModelCard.load(model.name_or_path)
+        else:
+            _logger.warning(
+                f"The version of huggingface_hub that is installed does not provide "
+                f"ModelCard functionality. You have version {hub.__version__} installed. "
+                f"Update huggingface_hub to >= '0.10.0' to retrieve the ModelCard data."
+            )
+    except ImportError:
+        _logger.warning(
+            "Unable to store ModelCard data with the saved artifact. In order to "
+            "preserve this information, please install the huggingface_hub package "
+            "by running 'pip install huggingingface_hub>0.10.0'"
+        )
```

Comment:
```suggestion
        import huggingface_hub as hub
    except ImportError:
        _logger.warning(
            "Unable to store ModelCard data with the saved artifact. In order to "
            "preserve this information, please install the huggingface_hub package "
            "by running 'pip install huggingingface_hub>0.10.0'"
        )
        return

    model = model_or_pipeline.model
    if hasattr(hub, "ModelCard"):
        return hub.ModelCard.load(model.name_or_path)
    else:
        _logger.warning(
            f"The version of huggingface_hub that is installed does not provide "
            f"ModelCard functionality. You have version {hub.__version__} installed. "
            f"Update huggingface_hub to >= '0.10.0' to retrieve the ModelCard data."
        )
```

Can we immediately exit if importing `huggingface_hub` fails?

Reply:
definitely! changed